protocol/communities/community_event.go

```diff
@@ -0,0 +1,461 @@
+	case protobuf.CommunityEvent_COMMUNITY_CHANNEL_REORDER:
+		_, err := o.reorderChat(communityEvent.ChannelData.CategoryId, communityEvent.ChannelData.ChannelId, int(communityEvent.ChannelData.Position))
+		if err != nil {
+			return err
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_CATEGORY_REORDER:
+		_, err := o.reorderCategories(communityEvent.CategoryData.CategoryId, int(communityEvent.CategoryData.Position))
+		if err != nil {
+			return err
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_ACCEPT:
+		for pkString, addedMember := range communityEvent.MembersAdded {
+			pk, err := common.HexToPubkey(pkString)
+			if err != nil {
+				return err
+			}
+			if !o.HasMember(pk) {
+				o.addCommunityMember(pk, addedMember)
+			}
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_REQUEST_TO_JOIN_REJECT:
+		for pkString := range communityEvent.RejectedRequestsToJoin {
+			pk, err := common.HexToPubkey(pkString)
+			if err != nil {
+				return err
+			}
+			o.removeMemberFromOrg(pk)
+		}
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_KICK:
+		pk, err := common.HexToPubkey(communityEvent.MemberToAction)
+		if err != nil {
+			return err
+		}
+
+		if o.IsMemberOwnerOrAdmin(pk) {
+			return errors.New("attempt to kick an owner or admin of the community from the admin side")
+		}
+
+		o.removeMemberFromOrg(pk)
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_BAN:
+		pk, err := common.HexToPubkey(communityEvent.MemberToAction)
+		if err != nil {
+			return err
+		}
+
+		if o.IsMemberOwnerOrAdmin(pk) {
+			return errors.New("attempt to ban an owner or admin of the community from the admin side")
+		}
+		o.banUserFromCommunity(pk)
+
+	case protobuf.CommunityEvent_COMMUNITY_MEMBER_UNBAN:
+		pk, err := common.HexToPubkey(communityEvent.MemberToAction)
+		if err != nil {
+			return err
+		}
+		o.unbanUserFromCommunity(pk)
+	}
+	return nil
+}
+
+func (o *Community) NewCommunityEventClock() uint64 {
+	return uint64(time.Now().Unix())
+}
+
+func (o *Community) addNewCommunityEvent(event *CommunityEvent) error {
+	err := validateCommunityEvent(event)
+	if err != nil {
+		return err
+	}
+
+	data, err := proto.Marshal(event.ToProtobuf())
+	if err != nil {
+		return errors.New("converting CommunityEvent to protobuf failed")
+	}
+
+	// All events must be build on top of the last owner CommunityDescription o the client side
+	// If there were no events before, extract CommunityDescription from MarshaledCommunityDescription
+	// and check the signature
+	if o.config.EventsData == nil || len(o.config.EventsData.EventsBaseCommunityDescription) == 0 {
+		metadata := &protobuf.ApplicationMetadataMessage{}
+
+		err := proto.Unmarshal(o.config.MarshaledCommunityDescription, metadata)
+		if err != nil {
+			return err
+		}
+
+		if metadata.Type != protobuf.ApplicationMetadataMessage_COMMUNITY_DESCRIPTION {
+			return ErrInvalidMessage
+		}
+
+		// TODO: check the signature
+		// Right now test are failing due to signer == nil, but this can be an issue in test
```

Comment:
Is this todo gonna be addressed in a later PR?

Reply:
yes, yesterday created the tickets